src/Symfony/Component/WebLink/Link.php

```diff
@@ -86,17 +103,21 @@ public function withHref(string|\Stringable $href): static
         return $that;
     }
 
-    public function withRel(string $rel): static
+    public function withRel(string|LinkRelation $rel): static
```

Comment:
Changing the argument type is not allowed by our [BC promise](https://symfony.com/doc/current/contributing/code/bc.html#changing-classes). I'm not sure about adding a type to the union type.

Reply:
In my opinion the addition of a typical union does not create a BC. Unit tests are always OK.

I wonder if this rule has taken into account the arrival of the union type.

If there is really a stop on this point I would go back with a constant class.

Reply:
As the method is not final, widening the allowed types for this parameter would break classes extending it: https://3v4l.org/bbFtY

Reply:
Thank you for this example.

I will come back with a proposal without BC as soon as possible